astropy/utils/data.py

```diff
@@ -1370,3 +1370,27 @@ def _release_download_cache_lock():
         msg = 'Error releasing lock. "{0}" either does not exist or is not ' +\
               'a directory.'
         raise RuntimeError(msg.format(lockdir))
+
+
+def _urls_on_shelf():
```

Comment:
Any reason for it to be private?  I don't see any reason not to make it public, actually! (i.e., no leading underscore)